src/NServiceBus.Core/Features/FeatureComponent.cs

```diff
@@ -0,0 +1,54 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Features;
+    using ObjectBuilder;
+    using Settings;
+
+    class FeatureComponent
+    {
+        public FeatureComponent(SettingsHolder settings)
+        {
+            this.settings = settings;
+        }
+
+        // concreteTypes will be some kind of TypeDiscoveryComponent when we encapsulate the scanning
+        public void Initalize(List<Type> concreteTypes, ContainerComponent containerComponent, FeatureConfigurationContext featureConfigurationContext)
+        {
+            featureActivator = new FeatureActivator(settings);
+
+            foreach (var type in concreteTypes.Where(t => IsFeature(t)))
+            {
+                featureActivator.Add(type.Construct<Feature>());
+            }
+
+            var featureStats = featureActivator.SetupFeatures(featureConfigurationContext);
+
+            builder = new Lazy<IBuilder>(() => containerComponent.Builder);
```

Comment:
What would be required to not need a `Lazy` here? Seems like a sign that some dependency layering isn't right yet.

Reply:
Yea this is something fishy here, the builder isn't valid for use at this point. I think it might be a sign that our ContainerComponent should be split into 2 => ContainerRegistrations + ContainerResolution. Our options that comes to mind is:

1. Keep the lazy
2. Store the entire component
3. Pass the component to the `.Start()` method instead 

I'm torn btw 1 and 3, what do you think?

Reply:
Since splitting the component is not in scope IMO

Reply:
The Lazy feels too much like a magical workaround to me, and passing a component into another component's Start method feels like a violation of the component concept, so I prefer option 2.